component-monitor/src/main.rs

```diff
@@ -0,0 +1,242 @@
+use anyhow::{bail, Context, Result};
+use log::{debug, error, info, warn, LevelFilter};
+use nix::sys::inotify::{AddWatchFlags, InitFlags, Inotify, InotifyEvent};
+use signal_hook::{
+    consts::{SIGINT, SIGTERM},
+    iterator::Signals,
+};
+use std::{fs, path::PathBuf, thread};
+use systemd_journal_logger::JournalLog;
+use walkdir::WalkDir;
+
+use clap::Parser;
+
+/// Manage the kernel-gpu-2404 contents
+#[derive(Debug, Parser)]
+#[command(version, about)]
+struct Cli {
+    /// The sentinel path to monitor
+    #[arg(env = "COMPONENT_SENTINEL_PATH")]
+    sentinel_path: PathBuf,
+    /// The target path to manage content in
+    #[arg(env = "COMPONENT_TARGET")]
+    target: PathBuf,
+    /// Turn on debug logging
+    #[arg(env, long, short)]
+    debug: bool,
+}
+
+/// Remove the sentinel file from target, then clear the target directory
+fn cleanup(target: &PathBuf, sentinel_name: &str) -> Result<()> {
+    info!(target: "files", "cleaning up {target:?}");
+    let sentinel_path = target.join(sentinel_name);
+    if sentinel_path.exists() {
+        if let Err(err) = fs::remove_file(&sentinel_path) {
+            warn!(target: "files", "failed to remove {:?}: {err}", &sentinel_path)
+        } else {
+            debug!(target: "files", "removed sentinel ({:?})", &sentinel_path)
+        }
+    }
+    for entry in fs::read_dir(target).context("failed to list target directory")? {
+        let entry = match entry {
+            Ok(e) => e,
+            Err(e) => {
+                error!(target: "files", "{e}");
+                continue;
+            }
+        };
+
+        let path = entry.path();
+        if path.is_dir() {
+            if let Err(err) = fs::remove_dir_all(&path) {
+                error!(target: "files", "failed to remove {:?}: {err}", &path)
+            } else {
+                debug!(target: "files", "removed {:?} recursively", &path)
+            }
+        } else {
+            if let Err(err) = fs::remove_file(&path) {
+                error!(target: "files", "failed to remove {:?}: {err}", &path)
+            } else {
+                debug!(target: "files", "removed {:?}", &path)
+            }
+        }
+    }
+
+    Ok(())
+}
+
+/// Check if the sentinel file is current, else clean the target directory and populate again,
+/// sentinel file being last
+fn populate(source: &PathBuf, sentinel_name: &str, target: &PathBuf) -> Result<()> {
+    info!(target: "files", "populating {target:?} from {source:?} with sentinel {sentinel_name:?}");
+    let sentinel_src = source.join(sentinel_name);
+
+    if !sentinel_src.exists() {
+        info!(target: "files", "sentinel file ({:?}) not found, skipping", &sentinel_src);
+        return Ok(());
+    }
+
+    let sentinel_data = match fs::read(&sentinel_src) {
+        Ok(d) if d.trim_ascii() == b"" => {
```

Comment:
I guess you can try `is_empty` instead  
```suggestion
        Ok(d) if d.is_empty() => {
```

Reply:
I don't want whitespace to be considered, hence the `trim_ascii()`.